test/Autofac.Test/Features/Scanning/OpenGenericScanningRegistrationTests.cs

```diff
@@ -119,6 +120,30 @@ public void AssignableToNullTypeProvidedThrowsException()
                 .AssignableTo(null, t => t));
         }
 
+        [Theory]
+        [InlineData(typeof(ICloseCommand))]
+        [InlineData(typeof(CloseCommand))]
+        public void AssignableToClosedTypeProvidedNoneOpenGenericSourceRegistered(Type closedType)
+        {
+            var cb = new ContainerBuilder();
+            cb.RegisterAssemblyOpenGenericTypes(typeof(ICommand<>).GetTypeInfo().Assembly)
+                .AssignableTo(closedType);
```

Comment:
It feels like passing a closed type into `AssignableTo` for Open Generics is enough of an input mistake that we should throw an `ArgumentException` if someone tries it?

Reply:
@alistairjevans  when using `AssignableTo` in registering assembly closed type, it doesn't throw exception when the argument is open generic type. Therefore should we keep this behavior in this case or we will modify it too ?

Reply:
Ah, good point. We should retain the same behaviour for consistency, and I don't think it's worth a breaking change in the existing `AssignableTo`.